stock_prices/stock_prices.py

```diff
@@ -1,10 +1,38 @@
 #!/usr/bin/python
 
 import argparse
+"""
+Problem:
+Find the maximum profits in a stock price pick
 
-def find_max_profit(prices):
-  pass
+Understand the problem:
+----------------------
+You can sell only in future
+you can not reverse back in transcation (can't go back in time)
+integers only
+you want to maximize the profits
+
+anlayze:
+--------
+"""
 
+def find_max_profit(prices):
+  #profit
+  max_value = 0
+  max_value_location = 0 #save the location of the max_value,we'll need it
+  min_value = 1e1000 #some artibrary big number to compare min_value against
+  # get the max value even the last element matters
+  for i in range(1, len(prices)):
```

Comment:
this is good! well done. another way to do this without two for loops is to keep track of the max profit while the for loop iterates through the first time.

min_price = prices[0]
max_profit = prices[1] - min_price

for i in range(1, len(prices)):
____price = prices[i]
____max_profit = max(price - min_price, max_profit)
____min_price = min(price, min_price)

return max_profit

Reply:
Hmm..right. Wouldn't be need to import math for this? I thought we weren't importing any such helper libraries. But thanks for sharing the tips! I'll try it this way.

Reply:
I believe max and min are built in methods to python so no, you wouldn't need to import math for this